scenarios/centos-9/tcib.yml

```diff
@@ -5,6 +5,7 @@ cifmw_build_containers_tcib_src: "{{ ansible_user_dir }}/src/github.com/openstac
 cifmw_repo_setup_src: "{{ ansible_user_dir }}/src/github.com/openstack-k8s-operators/repo-setup"
 
 cifmw_build_containers_repo_dir: "{{ cifmw_basedir }}/artifacts/repositories"
-cifmw_build_containers_push_registry: "{{ cifmw_rp_registry_ip }}:5001"
+cifmw_rp_registry_port: 5001
```

Comment:
do we really need to have this one here? If we do, then why not also cifmw_rp_registry_ip ?

They are both provided when we include the registry_deploy role https://github.com/openstack-k8s-operators/ci-framework/blob/0f6a0c1eba4e34660777065cff91a6e69c71e602/roles/registry_deploy/defaults/main.yml#L23-L24C1

or, are you already passing in cifmw_rp_registry_ip in the vars from somewhere else?

Reply:
We pass ` cifmw_rp_registry_ip` as a extra vars in[ tcib run playbook](https://github.com/openstack-k8s-operators/ci-framework/blob/main/ci/playbooks/tcib/run.yml#L28)
```
    - name: Run tcib playbook
      ansible.builtin.command:
        chdir: "{{ ansible_user_dir }}/src/github.com/openstack-k8s-operators/ci-framework"
        cmd: >-
          ansible-playbook ci/playbooks/tcib/tcib.yml
          -i "{{ ansible_user_dir }}/ci-framework-data/artifacts/zuul_inventory.yml"
          -e @scenarios/centos-9/base.yml
          -e @scenarios/centos-9/tcib.yml
          -e "cifmw_rp_registry_ip={{ node_ip }}"
          {%- if cifmw_extras is defined %}
          {%-   for extra_vars in cifmw_extras %}
          -e "{{   extra_vars }}"
          {%-   endfor %}
          {%- endif %}
          -e "@{{ ansible_user_dir }}/ci-framework-data/artifacts/parameters/zuul-params.yml"
```
Let me try removing cifmw_rp_registry_port from here and see what happens.

Reply:
After removal of cifmw_rp_registry_port, [cifmw-tcib job](https://logserver.rdoproject.org/00/1800/5d49406ceee9621864a728c51f9d4ec204c0caa4/github-check/cifmw-tcib/31a35b9/job-output.txt) is failing with `'cifmw_rp_registry_port'' is undefined.` at [Create artifacts with custom params](https://github.com/openstack-k8s-operators/ci-framework/blob/0ae35a6c86a9417b3fe89b9c7d08534222949c22/playbooks/01-bootstrap.yml#L63).

Since we are using `cifmw_rp_registry_port` var in scenario tcib var file. So it's default needs to be defined otherwise it will fail. The registry_deploy role default var will come into action once the role is called (which happens very late in the ansible playbook execution).

So, We need to define the default var for `cifmw_rp_registry_port` var.

Reply:
Thank you @marios for the feedback. I have  moved the port var as extra var to avoid confusion in the new update.